kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -31,47 +33,101 @@
 
 final class SchemaManagerImpl implements SchemaManager {
   private final SchemaRegistryClient schemaRegistryClient;
-  private final SubjectNameStrategy subjectNameStrategy;
 
   @Inject
-  SchemaManagerImpl(
-      SchemaRegistryClient schemaRegistryClient, SubjectNameStrategy subjectNameStrategy) {
+  SchemaManagerImpl(SchemaRegistryClient schemaRegistryClient) {
     this.schemaRegistryClient = requireNonNull(schemaRegistryClient);
-    this.subjectNameStrategy = requireNonNull(subjectNameStrategy);
   }
 
   @Override
-  public RegisteredSchema getSchemaById(EmbeddedFormat format, int schemaId) {
-    checkArgument(format.requiresSchema(), "%s does not support schemas.", format);
+  public RegisteredSchema getSchemaById(String subject, int schemaId) {
+    ParsedSchema schema;
+    try {
+      schema = schemaRegistryClient.getSchemaBySubjectAndId(subject, schemaId);
```

Comment:
whoa, i didn't know about this method, and don't understand why it exists. maybe change the name of your method to match `getSchemaBySubjectandIs`.

Reply:
This guarantees the subject has a version with this ID. Otherwise serialization would fail later.